app/models/concerns/wakes/metrics/facebook.rb

```diff
@@ -7,13 +7,26 @@ module Facebook
       included do
         store_accessor :document, :facebook_count, :facebook_count_updated_at
 
+        before_validation :prevent_update_of_legacy_location, on: :update
```

Comment:
@chrismaddocks what if, instead of all this, we redefined `#facebook_count=` to include a check for `canonical?` before setting the value? (And if not canonical, return some sort of warning.) Something like,

```
def facebook_count=(facebook_count)
  if canonical?
    super
  else
    'Cannot update the facebook count of a noncanonical location.'
  end
end
```